dome9/resource_dome9_admission_control_policy.go

```diff
@@ -0,0 +1,126 @@
+package dome9
+
+import (
+	"github.com/dome9/dome9-sdk-go/dome9/client"
+	"github.com/dome9/dome9-sdk-go/services/admissioncontrol/admission_policy"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/testing/variable"
+	"log"
+)
+
+func resourceAdmissionPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAdmissionControlPolicyCreate,
+		Read:   resourceAdmissionControlPolicyRead,
+		Update: resourceAdmissionControlPolicyUpdate,
+		Delete: resourceAdmissionControlPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"target_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"target_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Environment", "OrganizationalUnit"}, false),
+			},
+			"ruleset_id": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"notification_ids": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"action": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Prevention", "Detection"}, false),
+			},
+		},
+	}
+}
+
+func resourceAdmissionControlPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	d9Client := meta.(*Client)
+	req := expandAdmissionControlPolicyRequest(d)
+	log.Printf("[INFO] Creating Admission Control policy request %+v\n", req)
+	resp, _, err := d9Client.admissionControlPolicy.Create(&req)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Created admission control Policy with ID: %v\n", resp.ID)
+	d.SetId(resp.ID)
+
+	return resourceAdmissionControlPolicyRead(d, meta)
+}
+
+func expandAdmissionControlPolicyRequest(d *schema.ResourceData) admission_policy.AdmissionControlPolicyRequest {
+	targetType, isExists := d.GetOk("target_type")
+	if !isExists {
```

Comment:
this logic is not needed. the BE will set missing targetType to environment. no need to do it here too

Reply:
ok i found another way to do this even BE verify it, maybe on sdk will do some issue, i learned that the optional attribute inside tf schema can have default attribute, you can configure for it a function to bring dynamic value or to get it as a string, i configured it now as : 
"target_type": {
				Type:         schema.TypeString,
				Optional:     true,
				Default:      variable.AdmissionControlPolicyTargetType,